ksqldb-engine/src/main/java/io/confluent/ksql/execution/pull/HARouting.java

```diff
@@ -280,23 +235,20 @@ static PartitionFetchResult executeOrRouteQuery(
       final RoutingOptions routingOptions,
       final Optional<PullQueryExecutorMetrics> pullQueryMetrics,
       final PullPhysicalPlan pullPhysicalPlan,
-      final LogicalSchema outputSchema,
-      final QueryId queryId,
-      final PullQueryQueue pullQueryQueue,
-      final CompletableFuture<Void> shouldCancelRequests,
-      final Optional<ConsistencyOffsetVector> consistencyOffsetVector
+      final PullQueryWriteStream pullQueryQueue,
+      final CompletableFuture<Void> shouldCancelRequests
   ) {
-    final BiFunction<List<?>, LogicalSchema, PullQueryRow> rowFactory = (rawRow, schema) ->
-        new PullQueryRow(rawRow, schema, Optional.ofNullable(
-            routingOptions.getIsDebugRequest() ? node : null), Optional.empty());
+    final Function<StreamedRow, StreamedRow> addHostInfo
+        = sr -> sr.withSourceHost(routingOptions.getIsDebugRequest() ? toKsqlHostInfo(node) : null);
```

Comment:
tons of logic was moved out of `HARouting` to make the asynchronous nature of this PR a reality. The only reason it really needed to be in this class was because we added information to the response that included which host we read it from. Instead, I just extracted that logic into a function and passed that all the way down to `KsqlTarget`